src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -245,23 +258,46 @@ public async Task ExecuteCommandAsync(ListPackageArgs listPackageArgs)
         /// from different sources </param>
         /// <param name="listPackageArgs">Arguments for list package to get the right latest version</param>
         private void GetVersionsFromDict(IEnumerable<FrameworkPackages> packages,
-            Dictionary<string, IList<NuGetVersion>> packagesVersionsDict,
+            Dictionary<string, IList<IPackageSearchMetadata>> packagesVersionsDict,
             ListPackageArgs listPackageArgs)
         {
             foreach (var frameworkPackages in packages)
             {
                 foreach (var topLevelPackage in frameworkPackages.TopLevelPackages)
                 {
                     var matchingPackage = packagesVersionsDict.Where(p => p.Key.Equals(topLevelPackage.Name, StringComparison.OrdinalIgnoreCase)).First();
-                    topLevelPackage.LatestVersion = matchingPackage.Value.Where(newVersion => MeetsConstraints(newVersion, topLevelPackage, listPackageArgs)).Max();
-                    topLevelPackage.UpdateLevel = GetUpdateLevel(topLevelPackage.ResolvedVersion, topLevelPackage.LatestVersion);
+                    var latestVersion = matchingPackage.Value.Where(newVersion => MeetsConstraints(newVersion.Identity.Version, topLevelPackage, listPackageArgs)).Max(i => i.Identity.Version);
+
+                    topLevelPackage.LatestVersion = matchingPackage.Value.First(p => p.Identity.Version == latestVersion);
+                    topLevelPackage.UpdateLevel = GetUpdateLevel(topLevelPackage.ResolvedVersion.Identity.Version, topLevelPackage.LatestVersion.Identity.Version);
+
+                    // Update resolved version with deprecated information returned by the server.
+                    var resolvedVersionFromServer = matchingPackage.Value
+                        .Where(v => v.Identity.Version == topLevelPackage.ResolvedVersion.Identity.Version && v.DeprecationMetadata != null)
```

Comment:
This feels a bit unpredictable when multiple sources have deprecation information.
It's impossible right now, but it's information that will need to be surface eventually. 

I'd recommend discussing the experience with @anangaur and coming up with a plan. 

I'm ok if we say, this is not part of this change, but will be addressed in the future, as long as we have a plan.

Reply:
I agree. Instead of looking for the `.First()` entry, I'd suggest we take the one that has deprecation info available, if any, or fallback to `.First()` if none.

I think we need to come up with UX design when sources should be displayed, which is a non-trivial change I'd leave out of MVP for now.

@anangaur thoughts?

Reply:
This is a corner case scenario. If there are multiple sources with deprecation info for one or more package, I feel showing the first one if fine (assuming that is the behavior for showing the updates or package details). Additionally put out a message, something like:

```
There were multiple sources with deprecation information for one or more packages. Please use --source <source> parameter to view specific deprecation information from a given source.
```

Reply:
Tracked by https://github.com/NuGet/Home/issues/8372. We can continue the discussion there :)